File: cpp/example_plugin.cpp
#include "example_plugin.h"

ExamplePlugin::ExamplePlugin() : BasePlugin() {
}

bool ExamplePlugin::initialize(const PluginConfig& config) {
    bool result = BasePlugin::initialize(config);
    
    // 设置插件描述
    info.setDescription("这是一个示例插件");
    
    // 添加支持的命令
    info.addSupportedCommand("hello");
    info.addSupportedCommand("echo");
    
    // 添加支持的事件
    info.addSupportedEvent("startup");
    
    return result;
}

CommandResult ExamplePlugin::executeCommand(const std::string& command, const std::map<std::string, std::string>& params) {
    if (command == "hello") {
        return CommandResult(true, "Hello, World!", "");
    } else if (command == "echo") {
        auto it = params.find("message");
        std::string message = (it != params.end()) ? it->second : "";
        return CommandResult(true, "Echo: " + message, "");
    } else {
        return CommandResult(false, "", "未知命令: " + command);
    }
}

std::string ExamplePlugin::handleMessage(const std::string& message) {
    return "已收到消息: " + message;
}

File: rust/src/example_plugin.rs
use async_trait::async_trait;
use std::collections::HashMap;

use crate::base_plugin::BasePlugin;
use crate::command_result::CommandResult;
use crate::plugin_config::PluginConfig;
use crate::plugin_sdk::PluginSDK;

/// 示例插件实现
pub struct ExamplePlugin {
    base: BasePlugin,
}

impl ExamplePlugin {
    pub fn new() -> Self {
        Self {
            base: BasePlugin::new(),
        }
    }
}

#[async_trait]
impl PluginSDK for ExamplePlugin {
    async fn initialize(&mut self, config: PluginConfig) -> bool {
        let result = self.base.initialize(config).await;
        
        // 设置插件描述
        let mut info = self.base.get_info();
        info.set_description("这是一个示例插件".to_string());
        
        // 添加支持的命令
        info.add_supported_command("hello".to_string());
        info.add_supported_command("echo".to_string());
        
        // 添加支持的事件
        info.add_supported_event("startup".to_string());
        
        result
    }

    async fn start(&mut self) -> bool {
        self.base.start().await
    }

    async fn stop(&mut self) -> bool {
        self.base.stop().await
    }

    fn get_info(&self) -> crate::plugin_info::PluginInfo {
        self.base.get_info()
    }

    async fn execute_command(&self, command: &str, params: &HashMap<String, String>) -> CommandResult {
        match command {
            "hello" => CommandResult::new(
                true,
                "Hello, World!".to_string(),
                String::new(),
            ),
            "echo" => {
                let message = params.get("message").cloned().unwrap_or_default();
                CommandResult::new(
                    true,
                    format!("Echo: {}", message),
                    String::new(),
                )
            }
            _ => CommandResult::new(
                false,
                String::new(),
                format!("未知命令: {}", command),
            ),
        }
    }

    async fn handle_message(&self, message: &str) -> String {
        format!("已收到消息: {}", message)
    }
}